test/Interop/Cxx/templates/function-template-objc-typechecker.swift

```diff
@@ -0,0 +1,13 @@
+// RUN: not %target-swift-frontend -typecheck %s -I %S/Inputs -enable-experimental-cxx-interop 2>&1 | %FileCheck %s
+
+// REQUIRES: objc_interop
+
+import FunctionTemplates
+import ObjectiveC
+
+// Verify that ObjCBool type is banned from C++ template parameters.
+
+takesValue(ObjCBool(true))
+// CHECK: error: could not generate C++ types from the generic Swift types provided; the following Swift type(s) provided to 'takesValue' could not be converted: ObjCBool
```

Comment:
Is there a way to customize this error message for this case? Can we emit a fix-it that says use `.boolValue` to convert to a Swift boolean?

Reply:
Emitting a fix-it seems like a bit like an overkill for this case, I think. We would also need to handle possible explicit generic parameters (e.g. `myCxxMethod<T1, ObjCBool, T2>`). 
This bit of logic also doesn't have access to source locations. We could certainly pipe them through, but I'm not sure if it's worth the effort.